packages/consensus/src/types/utxo_set.cairo

```diff
@@ -32,10 +33,18 @@ pub impl UtxoSetImpl of UtxoSetTrait {
     }
 
     fn add(ref self: UtxoSet, output: OutPoint) {
+        let outpoint_hash = PoseidonTrait::new().update_with(output).finalize();
         if output.data.cached {
-            let outpoint_hash = PoseidonTrait::new().update_with(output).finalize();
             self.cache.insert(outpoint_hash, true);
-        } else { // TODO: update utreexo roots
+        } else {
+            self.utreexo_state.add(outpoint_hash);
```

Comment:
The idea is that `leaves_to_add` is an Array within the UtxoSet, so you do `self.leaves_to_add.append(outpoint_hash)`